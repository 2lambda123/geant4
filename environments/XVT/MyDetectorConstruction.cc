//
// ********************************************************************
// * DISCLAIMER                                                       *
// *                                                                  *
// * The following disclaimer summarizes all the specific disclaimers *
// * of contributors to this software. The specific disclaimers,which *
// * govern, are listed with their locations in:                      *
// *   http://cern.ch/geant4/license                                  *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.                                                             *
// *                                                                  *
// * This  code  implementation is the  intellectual property  of the *
// * GEANT4 collaboration.                                            *
// * By copying,  distributing  or modifying the Program (or any work *
// * based  on  the Program)  you indicate  your  acceptance of  this *
// * statement, and all its terms.                                    *
// ********************************************************************
//
//
// $Id: MyDetectorConstruction.cc,v 1.3 2001-07-11 09:56:47 gunter Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#include "MyDetectorConstruction.hh"

#include "MyCalorimeterSD.cc"
#include "MyCalorimeterHit.cc"
#include "MyCalorimeterHitsCollection.cc"

#include "G4Material.hh"
#include "G4MaterialTable.hh"
#include "G4Element.hh"
#include "G4ElementTable.hh"
#include "G4Box.hh"
#include "G4Tubs.hh"
#include "G4LogicalVolume.hh"
#include "G4ThreeVector.hh"
#include "G4PVPlacement.hh"
#include "G4SDManager.hh"
#include "G4VisAttributes.hh"
#include "G4Colour.hh"

MyDetectorConstruction::MyDetectorConstruction()
{;}

MyDetectorConstruction::~MyDetectorConstruction()
{;}

G4VPhysicalVolume* MyDetectorConstruction::Construct()
{
  G4cout << "Calorimeter volume Performance Test" << G4endl;

//--------- Material definition ---------

  G4double a, iz, z, density;
  G4String name, symbol;
  G4int nel;

  a = 14.01*g/mole;
  G4Element* elN = new G4Element(name="Nitrogen", symbol="N", iz=7., a);
  a = 16.00*g/mole;
  G4Element* elO = new G4Element(name="Oxigen", symbol="O", iz=8., a);

  a = 26.98*g/mole;
  density = 2.7*g/cm3;
  G4Material* Al = new G4Material(name="Aluminium", z=13., a, density);
  a = 207.19*g/mole;
  density = 11.35*g/cm3;
  G4Material* Pb = new G4Material(name="Lead", z=82., a, density);
  density = 1.29e-03*g/cm3;
  G4Material* Air = new G4Material(name="Air", density, nel=2);
  Air->AddElement(elN, .7);
  Air->AddElement(elO, .3);

//--------- G4VSolid, G4LogicalVolume, G4VPhysicalVolume  ---------

  G4double offset=22.5*cm, xTlate, yTlate;
  G4int i,j,copyNo;

  G4Box *myWorldBox= new G4Box("WBox",2000*cm, 2000*cm, 2000*cm);
  G4Box *myCalBox = new G4Box("CBox",1500*cm, 1500*cm, 1000*cm);
  G4Tubs *myTargetTube 
     = new G4Tubs("TTube",0*cm, 22.5*cm, 1000*cm, 0.*deg, 360.*deg);

  G4LogicalVolume *myWorldLog=new G4LogicalVolume(myWorldBox,Air,
						    "WLog", 0, 0, 0);
  G4LogicalVolume *myCalLog=new G4LogicalVolume(myCalBox,Al,
						  "CLog", 0, 0, 0);
  G4LogicalVolume *myTargetLog=new G4LogicalVolume(myTargetTube,Pb,
						     "TLog", 0, 0, 0);

  G4PVPlacement *myWorldPhys=new G4PVPlacement(0,G4ThreeVector(),
						 "WPhys",
						 myWorldLog,
						 0,false,0);
  G4PVPlacement *myCalPhys=new G4PVPlacement(0,G4ThreeVector(),
					       "CalPhys",
					       myCalLog,
					       myWorldPhys,false,0);

  G4String tName1("TPhys1");	// Allow all target physicals to share
				// same name (delayed copy)
  copyNo=0;
  for (j=1;j<=25;j++)
  {
    yTlate = -1000.0*cm - 40.0*cm + j*80.0*cm;
    for (i=1;i<=50;i++)
    {
      xTlate = -1000.0*cm - 20.0*cm + i*45.0*cm - offset;
      G4PVPlacement *myTargetPhys
        =new G4PVPlacement(0,G4ThreeVector(xTlate,yTlate,0*cm),
                           tName1,myTargetLog,myCalPhys,false,copyNo++);
    }
  }
  for (j=1;j<=26;j++)
  {
    yTlate = -1000.0*cm - 80.0*cm + j*80.0*cm;
    for (i=1;i<=50;i++)
    {
      xTlate = -1000.0*cm - 20.0*cm + i*45.0*cm;
      G4PVPlacement *myTargetPhys
        =new G4PVPlacement(0,G4ThreeVector(xTlate,yTlate,0*cm),
                           tName1,myTargetLog,myCalPhys,false,copyNo++);
    }
  }

//--------- Sensitive detector -------------------------------------

  G4SDManager* SDman = G4SDManager::GetSDMpointer();

  G4String calorimeterSDname = "example4/calorimeter";
  MyCalorimeterSD * myCalorimeterSD = new MyCalorimeterSD( calorimeterSDname );
  SDman->AddNewDetector( myCalorimeterSD );
  myTargetLog->SetSensitiveDetector(myCalorimeterSD);

//--------- Visualization attributes -------------------------------

  G4VisAttributes * experimantalHallVisAtt
    = new G4VisAttributes(G4Colour(1.0,1.0,1.0));
  experimantalHallVisAtt->SetVisibility(false);
  myWorldLog->SetVisAttributes(experimantalHallVisAtt);

  G4VisAttributes * calorimeterBoxVisAtt
    = new G4VisAttributes(G4Colour(0.0,0.0,0.5));
  calorimeterBoxVisAtt->SetForceWireframe(true);
  myCalLog->SetVisAttributes(calorimeterBoxVisAtt);

  G4VisAttributes * calorimeterTubeVisAtt
    = new G4VisAttributes(G4Colour(0.0,0.5,0.5));
  //  calorimeterTubeVisAtt->SetForceWireframe(true);
  myTargetLog->SetVisAttributes(calorimeterTubeVisAtt);

//------------------------------------------------------------------

  return myWorldPhys;
}

