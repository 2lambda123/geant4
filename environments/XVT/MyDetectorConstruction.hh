// This code implementation is the intellectual property of
// the RD44 GEANT4 collaboration.
//
// By copying, distributing or modifying the Program (or any work
// based on the Program) you indicate your acceptance of this statement,
// and all its terms.
//
// $Id: MyDetectorConstruction.hh,v 1.1 1999-01-07 16:04:59 gunter Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#ifndef MyDetectorConstruction_h
#define MyDetectorConstruction_h 1

#include "G4VUserDetectorConstruction.hh"
#include "globals.hh"

class MyDetectorConstruction : public G4VUserDetectorConstruction
{
  public:
    MyDetectorConstruction();
    ~MyDetectorConstruction();

  public:
     G4VPhysicalVolume* Construct();

};

#endif

