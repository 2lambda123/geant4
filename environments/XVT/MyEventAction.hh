// This code implementation is the intellectual property of
// the RD44 GEANT4 collaboration.
//
// By copying, distributing or modifying the Program (or any work
// based on the Program) you indicate your acceptance of this statement,
// and all its terms.
//
// $Id: MyEventAction.hh,v 1.1 1999-01-07 16:05:00 gunter Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#ifndef MyEventAction_h
#define MyEventAction_h 1

#include "G4UserEventAction.hh"
#include "globals.hh"

class G4Event;

class MyEventAction : public G4UserEventAction
{
  public:
    MyEventAction();
    ~MyEventAction();

  public:
    void BeginOfEventAction();
    void EndOfEventAction();

  private:
    G4int calorimeterCollID;
    /////////G4int bulkCollID;
    G4bool drawFlag;

  public:
    inline void SetDrawFlag(G4bool val)
    { drawFlag = val; };
};

#endif

    
