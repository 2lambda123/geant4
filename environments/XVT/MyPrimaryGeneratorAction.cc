// This code implementation is the intellectual property of
// the GEANT4 collaboration.
//
// By copying, distributing or modifying the Program (or any work
// based on the Program) you indicate your acceptance of this statement,
// and all its terms.
//
// $Id: MyPrimaryGeneratorAction.cc,v 1.2 1999-12-15 14:48:45 gunter Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#include "MyPrimaryGeneratorAction.hh"

#include "G4Event.hh"
#include "G4ParticleGun.hh"
#include "G4ParticleTable.hh"
#include "G4ParticleDefinition.hh"
#include "globals.hh"

MyPrimaryGeneratorAction::MyPrimaryGeneratorAction()
{
  G4int n_particle = 1;
  particleGun = new G4ParticleGun(n_particle);

// default particle

  G4ParticleTable* particleTable = G4ParticleTable::GetParticleTable();
  G4String particleName;
  G4ParticleDefinition* particle 
    = particleTable->FindParticle(particleName="e-");
  particleGun->SetParticleDefinition(particle);
  particleGun->SetParticleMomentumDirection(G4ThreeVector(1.,0.,0.));
  particleGun->SetParticleEnergy(1.*GeV);
  particleGun->SetParticlePosition(G4ThreeVector(0.*cm,0.*cm,0.*cm));

}

MyPrimaryGeneratorAction::~MyPrimaryGeneratorAction()
{
  delete particleGun;
}

void MyPrimaryGeneratorAction::GeneratePrimaries(G4Event* anEvent)
{
  particleGun->GeneratePrimaryVertex(anEvent);
}


