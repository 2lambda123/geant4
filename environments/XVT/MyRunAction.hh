// This code implementation is the intellectual property of
// the GEANT4 collaboration.
//
// By copying, distributing or modifying the Program (or any work
// based on the Program) you indicate your acceptance of this statement,
// and all its terms.
//
// $Id: MyRunAction.hh,v 1.2 1999-12-15 14:48:45 gunter Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#ifndef MyRunAction_h
#define MyRunAction_h 1

#include "G4UserRunAction.hh"
#include "G4Timer.hh"
#include "globals.hh"

class G4Run;

class MyRunAction : public G4UserRunAction
{
  public:
    MyRunAction();
    ~MyRunAction();

  public:
    void BeginOfRunAction(G4Run* aRun);
    void EndOfRunAction(G4Run* aRun);

  private:
    G4Timer* timer;
    G4int runIDcounter;
};

#endif

