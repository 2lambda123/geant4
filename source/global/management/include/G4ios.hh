// This code implementation is the intellectual property of
// the GEANT4 collaboration.
//
// By copying, distributing or modifying the Program (or any work
// based on the Program) you indicate your acceptance of this statement,
// and all its terms.
//
// $Id: G4ios.hh,v 1.4 1999-11-23 15:00:04 gcosmo Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//
// 
// ---------------------------------------------------------------
// GEANT 4 class header file
//
// G4ios.hh
//
// ---------------------------------------------------------------
#ifndef included_G4ios
#define included_G4ios

#if defined(OO_DDL_TRANSLATION)
/*
 * stdlib needs to be included before iostream.h
 * during oddlx runs to work around a parser
 * problem of AIX ooddlx v4.0.2 with some versions of
 * AIX system header files.
 */
#include <stdlib.h>
#endif

#include "g4std/iostream"
#include "G4Types.hh"

extern G4std::ostream G4cout;
extern G4std::ostream G4cerr;

#define G4cin G4std::cin
#define G4endl G4std::endl

#endif
