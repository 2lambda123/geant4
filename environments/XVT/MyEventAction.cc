// This code implementation is the intellectual property of
// the RD44 GEANT4 collaboration.
//
// By copying, distributing or modifying the Program (or any work
// based on the Program) you indicate your acceptance of this statement,
// and all its terms.
//
// $Id: MyEventAction.cc,v 1.1 1999-01-07 16:05:00 gunter Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#include "MyEventAction.hh"

#include "MyCalorimeterHit.hh"
#include "MyCalorimeterHitsCollection.hh"
#include <rw/tvordvec.h>

#include "G4Event.hh"
#include "G4EventManager.hh"
#include "G4HCofThisEvent.hh"
#include "G4VHitsCollection.hh"
#include "G4TrajectoryContainer.hh"
#include "G4Trajectory.hh"
#include "G4VVisManager.hh"
#include "G4SDManager.hh"
#include "G4UImanager.hh"
#include "G4ios.hh"

#include "MyEventActionMessenger.cc"

MyEventAction::MyEventAction()
:drawFlag(false)
{
  G4SDManager * SDman = G4SDManager::GetSDMpointer();
  G4String colNam;

  calorimeterCollID = SDman->GetCollectionID(colNam="CalCollection");
  //////////bulkCollID = SDman->GetCollectionID(colNam="bulkCollection");

  new MyEventActionMessenger(this);
}

MyEventAction::~MyEventAction()
{;}

void MyEventAction::BeginOfEventAction()
{
  if(drawFlag)
  {
    G4VVisManager* pVVisManager = G4VVisManager::GetConcreteInstance();
    if(pVVisManager)
    {
      G4UImanager::GetUIpointer()->ApplyCommand("/vis~/Draw/current");
    }
  }
}

void MyEventAction::EndOfEventAction()
{
  const G4Event* evt = fpEventManager->GetConstCurrentEvent();

  G4cout << ">>> Event " << evt->GetEventID() << endl;

  G4HCofThisEvent * HCE = evt->GetHCofThisEvent();
  MyCalorimeterHitsCollection* CHC = NULL;
  if(HCE)
    CHC = (MyCalorimeterHitsCollection*)(HCE->GetHC(calorimeterCollID));

  if(CHC)
  {
    RWTValOrderedVector<MyCalorimeterHit> theCollection
      = CHC->GetVector();
    int n_hit = theCollection.entries();
    G4cout << "     " << n_hit
         << " hits are stored in MyCalorimeterHitsCollection." << endl;
    G4double totE = 0;
    for(int i=0;i<n_hit;i++)
    { totE += theCollection[i].GetEdep(); }
    G4cout << "     Total energy deposition in calorimeter tubes : "
         << totE / GeV << " (GeV)" << endl;
  }

  if(drawFlag)
  {
    G4VVisManager* pVVisManager = G4VVisManager::GetConcreteInstance();
    if(pVVisManager)
    {
      if(CHC) CHC->DrawAllHits();
      G4UImanager::GetUIpointer()->ApplyCommand("/vis~/show/view");
    }
  }
}



