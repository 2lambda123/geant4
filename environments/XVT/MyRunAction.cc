// This code implementation is the intellectual property of
// the RD44 GEANT4 collaboration.
//
// By copying, distributing or modifying the Program (or any work
// based on the Program) you indicate your acceptance of this statement,
// and all its terms.
//
// $Id: MyRunAction.cc,v 1.1 1999-01-07 16:05:01 gunter Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#include "MyRunAction.hh"

#include "G4Run.hh"
#include "G4UImanager.hh"
#include "G4ios.hh"

MyRunAction::MyRunAction()
{
  timer = new G4Timer;
  runIDcounter = 0;
}

MyRunAction::~MyRunAction()
{
  delete timer;
}

void MyRunAction::BeginOfRunAction(G4Run* aRun)
{
  aRun->SetRunID(runIDcounter++);
  aRun->transient(true);

  G4cout << "### Run " << aRun->GetRunID() << " start." << endl;
  timer->Start();
}

void MyRunAction::EndOfRunAction(G4Run* aRun)
{
  timer->Stop();
  G4cout << "number of event = " << aRun->GetNumberOfEvent() 
       << " " << *timer << endl;
}

