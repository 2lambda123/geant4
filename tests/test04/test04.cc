// This code implementation is the intellectual property of
// the RD44 GEANT4 collaboration.
//
// By copying, distributing or modifying the Program (or any work
// based on the Program) you indicate your acceptance of this statement,
// and all its terms.
//
// $Id: test04.cc,v 1.1 1999-01-08 16:34:45 gunter Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//
//
// --------------------------------------------------------------
//      GEANT4 - test04
//
//      For information related to this code contact:
//      CERN, IT Division, ASD Group
// --------------------------------------------------------------
// Comments
//
//
// --------------------------------------------------------------


#include "MyDetectorConstruction.hh"
#include "MyPhysicsList.hh"
#include "MyRunAction.hh"
#include "MyPrimaryGeneratorAction.hh"
#include "MyEventAction.hh"
#include "MyStackingAction.hh"

#include "G4ios.hh"

#include "G4RunManager.hh"
#include "G4PersistencyManager.hh"

int main()
{
  // Set the default random engine to RanecuEngine
  RanecuEngine defaultEngine;
  HepRandom::setTheEngine(&defaultEngine);

  // Persistency Manager
  G4PersistencyManager * persistencyManager = new G4PersistencyManager;

  // Run manager
  G4RunManager * runManager = new G4RunManager;

  // Detector geometry
  runManager->SetUserInitialization(new MyDetectorConstruction);
  runManager->SetUserInitialization(new MyPhysicsList);
  
  // UserAction classes.
  runManager->SetUserAction(new MyRunAction);
  runManager->SetUserAction(new MyPrimaryGeneratorAction);
  runManager->SetUserAction(new MyEventAction);
  runManager->SetUserAction(new MyStackingAction);

  runManager->Initialize();

  // Event loop
  G4int n_event = 10;
  runManager->BeamOn(n_event);

  delete runManager;
  return 0;
}

