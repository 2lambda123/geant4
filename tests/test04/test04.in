#
# Macro file of "skeleton.cc"
# for tracking performance test
#
/control/verbose 2
# /run/verbose 2
#
# SimpleBox geometry made by Pb
# Geantino 100 GeV to the direction (1.,0.,0.)
# 10 events
#
/mydet/switchDetector SimpleBox
/mydet/selectMaterial Pb
/gun/particle geantino
/gun/energy 100 GeV
/gun/direction 1 0 0
/run/initialize
/run/beamOn 10
