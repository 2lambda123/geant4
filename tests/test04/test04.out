Unit category <Time> is not defined.
Opening Federated Database G4EXAMPLE.
  -- Tracker type sensitive detector

****************************************************************
****************************************************************
**                                                            **
**   G4MuPairProduction :                                     **
**      cross section/energy loss is calculated using         **
**      the accurate cross section formula of R.Kokoulin,     **
**      ( building the tables takes a lot of time,            **
**                   PLEASE BE PATIENT  ! )                   **
**      sampling of the energy of the pair is generated       **
**      according to the sampling tables.                     **
**                                                            **
****************************************************************
****************************************************************

****************************************************************
****************************************************************
**                                                            **
**   G4MuPairProduction :                                     **
**      cross section/energy loss is calculated using         **
**      the accurate cross section formula of R.Kokoulin,     **
**      ( building the tables takes a lot of time,            **
**                   PLEASE BE PATIENT  ! )                   **
**      sampling of the energy of the pair is generated       **
**      according to the sampling tables.                     **
**                                                            **
****************************************************************
****************************************************************
MyPhysicsList::setCuts:CutLength : 1 (mm)
>>> Event 0
Event : 0
mySD/EvenCollection
Number of hits          10
  Edep of the first hit 70.7118
mySD/OddCollection
Number of hits          48
  Edep of the first hit 44.5668
>>> Event 1
Event : 1
mySD/EvenCollection
Number of hits          36
  Edep of the first hit 22.3181
mySD/OddCollection
Number of hits          25
  Edep of the first hit 41.5939
>>> Event 2
Event : 2
mySD/EvenCollection
Number of hits          12
  Edep of the first hit 28.0978
mySD/OddCollection
Number of hits          52
  Edep of the first hit 3.52978
>>> Event 3
Event : 3
mySD/EvenCollection
Number of hits          14
  Edep of the first hit 5.96415
mySD/OddCollection
Number of hits          43
  Edep of the first hit 3.18537
>>> Event 4
Event : 4
mySD/EvenCollection
Number of hits          7
  Edep of the first hit 19.4871
mySD/OddCollection
Number of hits          57
  Edep of the first hit 7.21456
>>> Event 5
Event : 5
mySD/EvenCollection
Number of hits          6
  Edep of the first hit 1.65209
mySD/OddCollection
Number of hits          51
  Edep of the first hit 19.1148
>>> Event 6
Event : 6
mySD/EvenCollection
Number of hits          10
  Edep of the first hit 13.4296
mySD/OddCollection
Number of hits          47
  Edep of the first hit 62.0367
>>> Event 7
Event : 7
mySD/EvenCollection
Number of hits          6
  Edep of the first hit 36.4453
mySD/OddCollection
Number of hits          50
  Edep of the first hit 19.1307
>>> Event 8
Event : 8
mySD/EvenCollection
Number of hits          16
  Edep of the first hit 2.38804
mySD/OddCollection
Number of hits          52
  Edep of the first hit 48.9265
>>> Event 9
Event : 9
mySD/EvenCollection
Number of hits          12
  Edep of the first hit 19.3121
mySD/OddCollection
Number of hits          52
  Edep of the first hit 11.5742
